src/application-insights/azext_applicationinsights/_params.py

```diff
@@ -0,0 +1,39 @@
+        c.argument('top', help='The number of segments to return.  This value is only valid when segment is specified.')
+        c.argument('filter_arg', options_list=['--filter'], help=' An expression used to filter the results.  This value should be a valid OData filter expression where the keys of each clause should be applicable dimensions for the metric you are retrieving.')
+
+    with self.argument_context('monitor app-insights metrics get-metadata') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+
+    with self.argument_context('monitor app-insights events list') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+        c.argument('event_type', options_list=['--type'], help='The type of events to retrieve.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+
+    with self.argument_context('monitor app-insights events show') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+        c.argument('event_type', options_list=['--type'], help='The type of events to retrieve.')
+        c.argument('event', options_list=['--event'], help='GUID of the event to retrieve. This could be obtained by first listing and filtering events, then selecting an event of interest.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+
+    with self.argument_context('monitor app-insights query') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
```

Comment:
I believe this only accepts the full ARM resource ID(as well as the GUID). 
Please make it so that users can specify the resource-name and group using `--app` and `--resource-group`, which will be consistent with other commands.

Look here: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter
Using `id_part=` will also expose the `--ids` parameter for users to specify the full arm ID.